src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs

```diff
@@ -72,6 +78,13 @@ public sealed class RestoreManagerPackage : AsyncPackage
                 _buildEvents = dte.Events.BuildEvents;
                 _buildEvents.OnBuildBegin += BuildEvents_OnBuildBegin;
 
+                var solutionBuildManager = (IVsSolutionBuildManager5)await GetServiceAsync(typeof(SVsSolutionBuildManager));
+                if (solutionBuildManager != null)
```

Comment:
In general, yes assume it could be null. New SKUs, combinations of workloads, etc., or a package load failure for that service could all cause it to be null. You can use `Assumes.Present(solutionBuildManager)` for this, and it will throw an exception with a reasonably helpful error message if it is null.